### PR TITLE
[cli] Change wording on build warning for bundle id/package name

### PR DESCRIPTION
**Pull request by kbrandwijk**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

The warning about package name / bundle identifier also shows up after running `expo prebuild` or `expo run`, in which case it's expected that app.json contains these fields. The warning that it will be ignored once the native code exists is okay, but the deprecation notice should be removed.

# How

Old vs new:
image

# Test Plan

Tested by running `expo prebuild` followed by `eas build`.

### CODE REVIEW

**Comment by kbrandwijk** [comment]
/changelog-entry chore Change wording on build warning for bundle id/package name

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/project/android/applicationId.ts [hunk 1]
```diff
@@ -159,10 +159,10 @@ export function warnIfAndroidPackageDefinedInAppConfigForBareWorkflowProject(
 ): void {
   if (AndroidConfig.Package.getPackage(exp) && !warnPrinted) {
     Log.warn(
-      `Specifying "android.package" in ${getProjectConfigDescription(
+      `Specified value for "android.package" in ${getProjectConfigDescription(
         projectDir
-      )} is deprecated for bare workflow projects.\n` +
-        'EAS Build depends only on the value in the native code. Please remove the deprecated configuration.'
+      )} is ignored if native project folders are detected.\n` +
```
Comment (wkozyra95) [inline packages/eas-cli/src/project/android/applicationId.ts:164]: maybe 

```suggestion
      )} is ignored because native project folders were detected.\n` +
```

Also, this warning will be printed if the native directory exists and it is not gitignored, I think calling that bare workflow is more precise here, but I do not have a strong opinion on that

```suggestion
      )} is ignored because this is a bare workflow project.\n` +
```

This message has added benefit of pointing out to users that they have a bare project and a lot of them lately are ejecting unknowingly

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/project/android/applicationId.ts (hunk 1 above)
Comment (brentvatne) [inline packages/eas-cli/src/project/android/applicationId.ts:164]:
```suggestion
      )} is ignored because an ${chalk.bold('android')} directory was detected in the project, indicating that this is a bare workflow project.\n` +
```

a bit wordy but
- lets people know what triggered this detection (although we could possibly be more precise - detected dir and wasn't gitignored)
- factors in @wkozyra95's comment about bare workflow

Comment (kbrandwijk) [reply packages/eas-cli/src/project/android/applicationId.ts:164]: I specifically moved away from the old 'bare workflow' wording and opted for 'native project folders' because running prebuild is not the same as bare workflow. Bare workflow is where you have made manual changes to native files.

Comment (kbrandwijk) [reply packages/eas-cli/src/project/android/applicationId.ts]: See my comment above. I specifically moved away from saying 'bare workflow'. I do like the more specific folder naming.

Comment (wkozyra95) [reply packages/eas-cli/src/project/android/applicationId.ts]: I agree that running prebuild is not the same as bare workflow, but not in the way you described. Bare workflow is when you run prebuild and android/ios are not gitignored. Also, this issue is present only in bare workflow, not every time you run prebuild.

Comment (kbrandwijk) [reply packages/eas-cli/src/project/android/applicationId.ts]: If I init a project, define my identifiers, run `expo prebuild`, then `eas build`, I get this warning. One of the most common causes for confusion is that we advertise `expo prebuild` and `expo run` as something that can repeatedly be run _in the managed worfklow_, so we shouldn't mention 'bare workflow' here at all imho.
Bottom line here is: there's a native folder, so we're using it (unless you manually gitignore or easignore it, which is not by default in our templates). Mentioning 'bare workflow' without saying 'there's a native folder' is confusing for users, our daily support questions make that very clear.

**Comment by kbrandwijk** [comment]
image
Going with the middle road of the more specific wording.
